src/Extractor/CommonMetadataProvider.php

```diff
@@ -182,26 +181,18 @@ private function queryColumnsAndConstraints(?array $whitelist = null): array
         $sql[] = 'FROM INFORMATION_SCHEMA.COLUMNS as c';
         $sql[] = 'LEFT OUTER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE as kcu';
         $sql[] = 'ON c.TABLE_NAME = kcu.TABLE_NAME AND c.COLUMN_NAME = kcu.COLUMN_NAME';
-        $sql[] = sprintf('WHERE c.TABLE_SCHEMA = %s', $this->db->quote($this->database));
+        $sql[] = sprintf('WHERE c.TABLE_SCHEMA = %s', $this->connection->quote($this->database));
 
         if ($whitelist) {
-            $sql[] =sprintf('AND c.TABLE_NAME IN (%s)', $this->quoteTables($whitelist));
+            $sql[] = sprintf('AND c.TABLE_NAME IN (%s)', $this->quoteTables($whitelist));
         }
 
         $sql[] = ' ORDER BY c.TABLE_SCHEMA, c.TABLE_NAME, ORDINAL_POSITION';
 
         // Run query
-        return $this->queryAndFetchAll(implode(' ', $sql));
+        return $this->connection->query(implode(' ', $sql))->fetchAll();
     }
 
-    private function queryAndFetchAll(string $sql): array
```

Comment:
Tato metoda viac nie je potrebna, kedze existuje nova metoda `QueryResult::fetchAll`
- https://github.com/keboola/db-extractor-adapter/blob/c12f04421c0d73cde5a0bb3ef94547affc26e72e/src/ValueObject/QueryResult.php#L32
- https://github.com/keboola/db-extractor-adapter/blob/c12f04421c0d73cde5a0bb3ef94547affc26e72e/src/PDO/PdoQueryResult.php#L50-L55
- https://github.com/keboola/db-extractor-adapter/blob/c12f04421c0d73cde5a0bb3ef94547affc26e72e/src/ODBC/OdbcQueryResult.php#L53-L56

Reply:
yes, this is great